### PR TITLE
2nd Assignment by 201903774 한형준

### PR DESCRIPTION
**Pull request by hhj2000**
# MNIST 분류 실험 결과

## 기본 모델 성능
- 최종 테스트 정확도: 97.18%
- 훈련 시간: 0분 46초

### 실험 1: 하이퍼파라미터 튜닝
- 변경사항: 은닉층 크기 증가(hidden_size1: 100 -> 589), 배치 크기 증가(batch_size: 128 -> 138), 에포크 수 증가(nb_epochs: 3 -> 4)
- 결과: 최종 테스트 정확도 상승(97.06% -> 97.91%), 훈련 시간 증가(0분 47초 -> 1분)

### 실험 2: 모델 구조 개선
- 변경사항: 은닉층 2개 추가로 3층 신경망 구성(hidden_size1=589, hidden_size2=256, hidden_size3=128), LeakyReLU 및 Tanh 적용(전자는 첫 번째와 세 번째, 후자는 두 번째에 적용), Dropout 모든 층에 적용(p=0.3), BatchNormalization 1d 모든 층에 적용
- 결과: 최종 테스트 정확도 상승 및 변동 안정화(97.91% -> 98.01%), 훈련 시간 변동 없음(1분)

### CODE REVIEW

File: submissions/201903774/assignment2/experiments/hyperparameter_tuning.ipynb
[the remark this answers is not in the document]
Comment (hist0613) [reply submissions/201903774/assignment2/experiments/hyperparameter_tuning.ipynb:1]: 넵 독립적으로 실행되게 해주세요ㅎㅎ

[the remark this answers is not in the document]
Comment (hhj2000) [reply submissions/201903774/assignment2/experiments/hyperparameter_tuning.ipynb:1]: 알겠습니다 ㅋㅋ, 조금만 기다려주시면 바로 고쳐서 제출할께요!

[the remark this answers is not in the document]
Comment (hhj2000) [reply submissions/201903774/assignment2/experiments/hyperparameter_tuning.ipynb:1]: 방금 수정 완료했고 혹시 더 수정해야 할 부분 있는지 알려주시면 감사하겠습니다!

**Comment by hist0613** [comment]
돌린 실험들이 재현 가능하게 해주세요~
지금 상태로는 제가 시간을 꽤나 들여야 테스트를 해볼 수 있습니다

**Comment by hhj2000** [comment]
아아 알겠습니다 ㅎㅎ 그럼 과제 코드에 주석을 통해 실행 순서를 비롯한 설명을 좀 더 보충해서 수정해 올리면 될까요?

**Comment by hhj2000** [comment]
방금 수정 사항 반영해서 `experiments/` 폴더에 들어간 각 실험 코드를 독립적으로 실행 가능한 완결 버전으로 수정했고, `results/`에도 각 실험마다 산출된 도표들을 포함했습니다! 확인해보시고 아직 더 수정해야 할 부분이 있으면 답장 주시거나 내일 강의 쉬는 시간 혹은 끝나고 나서 피드백 주시면 감사하겠습니다!

**Comment by hist0613** [comment]
> 평가해주셔서 감사합니다! merge가 불가능하다면 제가 수정한 ml4e-mnist.ipynb 파일을 새로운 이름의 파일로 수정하고 기존의 ml4e-mnist.ipynb를 다시 pull해서 내려받는 식으로 하면 될까요, 아니면 Close pull request를 하면 좋을 지 말씀해주신다면 감사하겠습니다!

기존의 ml4e-mnist.ipynb 를 복구하시면 됩니다!
